rhine/src/Data/Automaton/MSF/Trans/Except.hs

```diff
@@ -0,0 +1,286 @@
+{-# LANGUAGE Arrows #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE Rank2Types #-}
+{-# LANGUAGE StrictData #-}
+
+{- | 'MSF's in the 'ExceptT' monad are monadic stream functions that can throw
+exceptions, i.e. return an exception value instead of a continuation. This
+module gives ways to throw exceptions in various ways, and to handle them
+through a monadic interface.
+-}
+module Data.Automaton.MSF.Trans.Except (
+  module Data.Automaton.MSF.Trans.Except,
+  module Control.Monad.Trans.Except,
+)
+where
+
+-- base
+import Control.Arrow (arr, returnA, (<<<), (>>>))
+import Control.Category qualified as Category
+import Control.Monad.Trans.Except hiding (
+  liftCallCC,
+  liftListen,
+  liftPass,
+ )
+import Control.Monad.Trans.Maybe (MaybeT, runMaybeT)
+import Data.Void (Void)
+
+-- Internal imports
+
+import Control.Monad.Morph
+import Control.Monad.Trans.Reader
+import Data.Automaton.Except hiding (safely)
+import Data.Automaton.Except qualified as AutomatonExcept
+import Data.Automaton.MSF (
+  MSF (..),
+  arrM,
+  constM,
+  count,
+  feedback,
+  hoistS,
+  liftS,
+  mapMaybeS,
+  reactimate,
+ )
+import Data.Automaton.Optimized (mapOptimizedAutomatonT)
+import Data.Automaton.Optimized qualified as AutomatonOptimized
+import Data.Either (fromLeft)
+
+-- FIXME I only need this API in ClSF
```

Comment:
Probably restrict to the bare necessities, rather port things to the ClSF modules.

Reply:
(This logic applies to all Trans modules)